### DISCUSSION TITLE
Art Engine Speed

### DISCUSSION THREAD

**Discussion by VeriumETH** [nftchef/art-engine#88] [Help + Q&A] [answered]
Hello!

I was just wondering if the resolution option affects speed or is it the configuration of having to sort through incompatibilities and forced combinations? I noticed that if I generate 10 @ 1080px it takes 95-99 seconds and the exact amount of time to generate 10 @ 256px

If anyone knows why or maybe I'm doing something wrong? I've used the original Hashlips art engine to compare and was able to generate 10 @ 512px in about 5 seconds. I would love to know so it doesn't take me 20-24 hours to generate my collection...If the program stops part way it would be very annoying for my project team. The point for our project is to not curate the collection but mint from random generation. Any help would be greatly appreciated!

**Comment by nftchef** [comment, marked as answer]
Hi, 
The generator is not really optimized for speed. Yes, the higher the resolution the longer it will take. 
If you have a lot of forced combinations and incompatible configurations, that will also significantly reduce performance (increase time per image). And, the more layers you add, the longer it will take to process, too. 

Using folder branching instead of incompatible and forced combinations will speed up the performance significantly.

**Comment by VeriumETH** [comment]
Thank you @nftchef 

That makes perfect sense. I will definitely explore the folder branching technique for comparison.
